linter.py

```diff
@@ -0,0 +1,39 @@
+import os
+import re
+from typing import Pattern, List
+
+search_note_pattern: Pattern = re.compile(r'^\*\*Note\*\*')
+search_warning_pattern: Pattern = re.compile(r'^\*\*Warning\*\*')
+correct_note_pattern: Pattern = re.compile(r'^> \*\*Note\*\*')
+correct_warning_pattern: Pattern = re.compile(r'^> \*\*Warning\*\*')
+
+def find_regex_in_file_content(
+    search_pattern: Pattern,
+    correct_pattern: Pattern,
+    filenames: List[str]
+) -> bool:
+    """Function that looks for a regex pattern in the given file contents."""
+
+    exit_code = 0
+    for filename in filenames:
+        with open(filename) as file:
+            lines = file.readlines()
+        for i, line in enumerate(lines):
+            if search_pattern.search(line):
+                print(f'{filename}:{i+1}: error: wrong notation "{search_pattern.pattern}"; expected "{correct_pattern.pattern}"')
+                exit_code = 1
```

Comment:
Could this be `match_found` instead since we don't actually use it as an exit code?

Reply:
Yeah, `match_found` sounds self-explanatory. I have updated the code accordingly